site/src/components/TemplateFiles/TemplateFiles.tsx

```diff
@@ -39,19 +39,22 @@ const languageByExtension: Record<AllowedExtension, string> = {
 
 interface TemplateFilesProps {
   currentFiles: TemplateVersionFiles;
-  previousFiles?: TemplateVersionFiles;
+  /**
+   * Files used to compare with current files
+   */
+  baseFiles?: TemplateVersionFiles;
   tab: UseTabResult;
 }
 
 export const TemplateFiles: FC<TemplateFilesProps> = ({
   currentFiles,
-  previousFiles,
+  baseFiles,
   tab,
 }) => {
   const filenames = Object.keys(currentFiles);
   const selectedFilename = filenames[Number(tab.value)];
   const currentFile = currentFiles[selectedFilename];
-  const previousFile = previousFiles && previousFiles[selectedFilename];
+  const previousFile = baseFiles && baseFiles[selectedFilename];
```

Comment:
Not a call out to your code, but there's some weird behavior here (based around `undefined`) that I'm guessing that the compiler isn't complaining about right now, because we don't have it set strictly enough.

Here's a [TypeScript snippet with the compiler tightened up](https://www.typescriptlang.org/play?noUncheckedIndexedAccess=true#code/MYewdgzgLgBMCuAnRBTMUBiBLANiiAXDAEoqiIAmAPNIlmAOYA0Mt9DAfDALwwDeAXwDcAKFCRYAIwCGEFNjyEY8MBRQAzeigowAPiTIhKNKHUYs2jLr0Gjx0GFGmSAatJzwUPGACIATD6iYuAOmnhg0gC2+N4A8pIAVmRQAHQA1igAnhAAFAjIaJi4+ACUdiGwcnjAUNoKaFFevGEN0RAA2gBy8JGSKIg5Tq7uniUAuuUScEio6PXe+bNFiu1VyXXFEdETwVMADqgAblgg8BDzvDJy9RAwAGR3MFfyxR1rNRvhjRNAA). I'm going to be bringing this up at the Frontend Variety later today

Reply:
Yes, I think we should set this rule as true: https://www.typescriptlang.org/docs/handbook/release-notes/typescript-4-1.html#checked-indexed-accesses---nouncheckedindexedaccess